packages/runtime-dom/src/directives/vModel.ts

```diff
@@ -233,7 +239,7 @@ export const vModelSelect: ModelDirective<HTMLSelectElement, 'number'> = {
     el[assignKey] = getModelAssigner(vnode)
   },
   updated(el, { value }) {
-    if (!el._assigning) {
+    if (!el[assigningKey] || toRaw(value) !== el[assignValueKey]) {
```

Comment:
Using an equality check might be problematic if the collection is modified, rather than replaced.

For example:

- [Playground](https://deploy-preview-12428--vue-sfc-playground.netlify.app/#eNp9Uk1v2zAM/SuELnGAzOmwnTI3QFP0sB22YRl2mXYwbCZVJ0uCPpwMhv/7KKlxkn30Ypjke3yPFAd2Z0zZB2QrVrnGCuPBoQ9mzZXojLYeBrC4W0CjOxM8tjDCzuoOZkSaccVVo5WLJIlNLN9GeKHwAFv0xXx+hnRBemEkXkBPTYuBK4A9eijmkP6B2vhg1dS47GsZMJbGRfySS4gy32o5UXJYtkTxWMw2M1KP+esepJuBqRlXI6GqZR6exqbAY2dk7ZEigKoV/Xr73GEFw/D3IONYLSMqkomQ89C/6jRZueXsTwJnU48kQRxtvNBqfVctn/+u85v/5O+v8jRFUqCgWl4MQaHzv5JYBnw/yf/IqzuI1j+u4PXNjTm+o42khSQGWzDv6P12Yl8+Oa3oThKDs/h2QqL9lPQdZ7SbbI6zWkp9+JBy3gZMD5Y4j9j8/Ef+yR1jjrPPlpZke+Rsqvna0mHk8sP2Ix7pfyrShoMk9AvFL+i0DNFjhm2Casn2BS65fZ+uXaj9V/dw9Kjcaaho9Hx2nNHd378w+tnum/Lt6cLY+BtjXyhP)
- [Playground](https://deploy-preview-12428--vue-sfc-playground.netlify.app/#eNp9Uk1PwzAM/StWLnTS6EBwGl0lhjjAARAfJ8Jhar0tI02iJO2Gqv533HTdhwS72c/v2S+Oa3ZrTFyVyMYscZkVxoNDX5qUK1EYbT3UYHE+BK2mONcWP0w+8wgNzK0u4IykZ1xxlWnlWqnEzGMOk1YUfX4N2tqxNIoGMEmh5gpgvRQSIeplcTWTJcYS1cIvIYXLQUeDXeMtw2gTUWuAhquGgmTUeSfXlHgsjKRJlAEkuajSt618DHW9N9k0yaittiIidjhU54XOUU4464mcQVFKL4zsWhJXGy+0Sm+T0TY6xqf/4HdHOLkOEyhJRgemKXX+JwzrCJ/9+K/t2kTul2O4vLgwmxvaQFhAULAh847+Yi4W8cppRd8aFJxlujC0bPsc5jvOaBedOc5mUur1Y8C8LXHY49kSs+8/8JXbtBhnLxYd2go529X8zC7Qd+X7tyfcULwr0mZLSewTxVd0Wpatx442LVVOtg94we1DOE6hFu/ufuNRuf5RrdFwGYHPGR3o3Ymn7+1exdf9RbHmF6xwC5Y=)

I'm not sure how important handling cases like these is. The change doesn't make them any worse.